src/aap_eda/core/models/credential.py

```diff
@@ -38,5 +38,6 @@ class Meta:
     )
     username = models.TextField(null=True)
     secret = EncryptedTextField(null=False)
+    key = models.TextField(null=True)
```

Comment:
I feel "key" is a little generic. Why not "variable_name" or "extra_var_key"?

Reply:
This value is mandatory depending on the credential_type. I think we should check this constraint for integrity. 
https://docs.djangoproject.com/en/4.2/ref/forms/validation/
https://docs.djangoproject.com/en/4.2/ref/validators/

Reply:
> This value is mandatory depending on the credential_type. I think we should check this constraint for integrity. https://docs.djangoproject.com/en/4.2/ref/forms/validation/ https://docs.djangoproject.com/en/4.2/ref/validators/

Yes, the validator will be added in the next PR

Reply:
We have a credential model which supports different types and each type has a different set of required attribute names, are we going to share some of the attribute names across these different types. Otherwise this table will have the attributes for all types in a single table

Reply:
"key" is changed to "identifier"